addons/hr/models/hr_employee.py

```diff
@@ -134,6 +134,56 @@ def _default_image(self):
         ('user_uniq', 'unique (user_id, company_id)', "A user cannot be linked to multiple employees in the same company.")
     ]
 
+    @api.multi
+    def name_get(self):
+        try:
+            return super(HrEmployeePrivate, self).name_get()
+        except AccessError:
+            return self.env['hr.employee.public'].browse(self.ids).name_get()
+
+    @api.multi
+    def read(self, fields, load='_classic_read'):
+        if self.check_access_rights('read', raise_exception=False):
+            return super(HrEmployeePrivate, self).read(fields, load=load)
+        else:
+            public_fields = set(self.env['hr.employee.public']._fields.keys()).intersection(fields)
+            if not public_fields:
+                raise AccessError(_('All the fields you try to access are private but you cannot read private fields on employees'))
+            # We fill all the fields with False as a value in order to stay coherent with what was asked
+            base = dict.fromkeys(fields, False)
+            return [{**base, **result} for result in self.env['hr.employee.public'].browse(self.ids).read(public_fields, load=load)]
```

Comment:
double-check this, could we avoid this magic and call the public employee only if all fields are public ones. Other cases should simply call super (and probably raise). Can't we simply fix the cases (views/actions) where this happens?